### PR TITLE
Adding support for diy pcie switch

### PR DESCRIPTION
**Pull request by Yuke-hd**

### CODE REVIEW

**Comment by Koenkk** [comment]
Could you provide a link of a how-to of this device?

**Comment by Yuke-hd** [comment]
lmk if this enough https://github.com/Yuke-hd/zigbee-pcie-switch

File: src/devices/custom_devices_diy.ts [hunk 1]
```diff
@@ -1167,6 +1168,53 @@ const definitions: Definition[] = [
...
+                    const payload = {ctrlbits: 0, ontime: Math.round(2), offwaittime: Math.round(2)};
+
+                    return await entity.command('genOnOff', 'onWithTimedOff', payload, utils.getOptions(meta.mapped, entity));
+                },
+            } satisfies Tz.Converter,
+        ],
+        exposes: [
+            e.binary('status', ea.STATE, true, false).withDescription('Indicates if the PC is powered on (= true) or off (= false)'),
+            e.enum('Power', ea.STATE_SET, ['press']).withDescription('Power button').withEndpoint('power'),
+            e.enum('Reset', ea.STATE_SET, ['press']).withDescription('Reset button').withEndpoint('reset'),
```
Comment (Koenkk) [inline src/devices/custom_devices_diy.ts:1210]:
```suggestion
            e.enum('reset', ea.STATE_SET, ['press']).withDescription('Reset button').withEndpoint('reset'),
```

File: src/devices/custom_devices_diy.ts [hunk 2]
```diff
@@ -1167,6 +1168,53 @@ const definitions: Definition[] = [
...
+                convertSet: async (entity, key, value, meta) => {
+                    const payload = {ctrlbits: 0, ontime: Math.round(2), offwaittime: Math.round(2)};
+
+                    return await entity.command('genOnOff', 'onWithTimedOff', payload, utils.getOptions(meta.mapped, entity));
+                },
+            } satisfies Tz.Converter,
+        ],
+        exposes: [
+            e.binary('status', ea.STATE, true, false).withDescription('Indicates if the PC is powered on (= true) or off (= false)'),
+            e.enum('Power', ea.STATE_SET, ['press']).withDescription('Power button').withEndpoint('power'),
```
Comment (Koenkk) [inline src/devices/custom_devices_diy.ts:1209]:
```suggestion
            e.enum('power', ea.STATE_SET, ['press']).withDescription('Power button').withEndpoint('power'),
```

File: src/devices/custom_devices_diy.ts [hunk 3]
```diff
@@ -1167,6 +1168,53 @@ const definitions: Definition[] = [
             }),
         ],
     },
+    {
+        zigbeeModel: ['ESP32C6.PCIE-switch'],
+        model: 'ESP32C6.PCIE-switch',
+        vendor: 'Custom devices (DiY)',
+        description: 'PCIE switch',
+        fromZigbee: [
```
Comment (Koenkk) [inline src/devices/custom_devices_diy.ts:1176]: Please move the `fromZigbee`/`toZigbee` converter to `tzLocal`/`fzLocal` (example: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/adeo.ts#L11)

**Comment by Koenkk** [comment]
Could you also submit a picture for the docs, also include the link to your github in the `.md` file of this device. This can be done by:
1. Create a fork by clicking [here](https://github.com/Koenkk/zigbee2mqtt.io/fork)
2. Go to the `public/images/devices` directory, *Add file* -> *Upload files*
3. Upload the files and press *Commit changes*
4. Press *Contribute* -> *Open pull request* -> update title/description -> *Create pull request*

Make sure that:
- The filename is `ESP32C6.PCIE-switch.png`
- The size is 512x512
- The background is transparent (use e.g. [Adobe remove background](https://new.express.adobe.com/tools/remove-background))

**Pull request closed without merging**